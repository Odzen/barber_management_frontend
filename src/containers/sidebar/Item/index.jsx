import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import './item.scss'

/* Component to display each of the modules in the Sidebar */

const Item = ({ text, to, svg, open, classes }) => {


  return (
    <div className='container_item'>
      <NavLink className={classes} to={to}>
        <div>{svg}</div>
        <p>{text}</p>
      </NavLink>
    </div>
  )
}

Item.propTypes = {
  text: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  svg: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired
}

export default Item
