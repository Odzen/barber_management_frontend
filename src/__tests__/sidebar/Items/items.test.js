import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Item from '../../../containers/sidebar/Item/index'

afterEach(() => {
  cleanup()
})

describe('testing to Item component', () => {
  test('should return Item component', () => {
    let text = 'Usuarios'
    let to = '/users'
    let svg = ''
    let open = true
    let classes = open ? 'linkOpen' : 'normal'

    let item = renderer
      .create(
        <MemoryRouter initialEntries={['/users/mjackson']}>
          <Item text={text} to={to} svg={svg} open={open} classes={classes} />
        </MemoryRouter>
      )
      .toJSON()

    expect(item).toMatchSnapshot()

  })
})
