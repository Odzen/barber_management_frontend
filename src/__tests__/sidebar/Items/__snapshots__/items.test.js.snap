// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing to Item component should return Item component 1`] = `
<div
  className="container_item"
>
  <a
    aria-current="page"
    className="linkOpen active"
    href="/users"
    onClick={[Function]}
  >
    <div />
    <p>
      Usuarios
    </p>
  </a>
</div>
`;
